src/main/kotlin/interactor/GetCitiesNamesSortedCheapestBananaPrices.kt

```diff
@@ -1,18 +1,21 @@
 package interactor
 
 import model.CityEntity
+import utils.isNotNull
+import java.lang.IllegalStateException
 
 class GetCitiesNamesSortedCheapestBananaPrices {
 
     fun execute(vararg cities:CityEntity):List<String>{
         return cities
-            .filter(::excludeNullCityAndLowQualityData)
+            .also { it.ifEmpty { throw NoSuchElementException("The list of cities you entered is empty") } }
```

Comment:
You can use **ifEmpty** function instead of also:
```suggestion
    fun execute(vararg cities:CityEntity):List<String>{
        return cities
            .ifEmpty { throw IllegalStateException("The list of cities you entered is empty") }
            .filter(::excludeNullCityAndLowQualityData)
            .sortedBy { it.fruitAndVegetablesPrices.banana1kg }
            .map { it.cityName }
    }```